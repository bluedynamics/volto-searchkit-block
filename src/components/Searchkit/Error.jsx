import { useSelector } from 'react-redux';
import { Header, Segment } from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';

const Error = () => {
  const error = useSelector((state) => state.results?.error);

  return (
    <Segment>
      <Header icon>
        <FormattedMessage
          id="Check the configuration of your searchkit block!"
          defaultMessage="Check the configuration of your searchkit block!"
        />
      </Header>
      <i>{error?.message}</i>
    </Segment>
  );
};

export default Error;
