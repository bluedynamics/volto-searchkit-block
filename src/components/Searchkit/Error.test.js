import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import Error from './Error';

beforeAll(() => {});

const mockStore = configureStore();

describe('Generic Error', () => {
  it('renders a simple error component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
      results: {
        error: {
          name: 'NotFoundError',
          message: 'Index cannot be found.',
        },
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Error error={{ status: 401, message: 'Unauthorized' }} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
