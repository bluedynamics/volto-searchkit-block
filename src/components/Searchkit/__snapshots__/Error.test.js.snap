// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Error renders a simple error component 1`] = `
<div
  className="ui segment"
>
  <div
    className="ui icon header"
  >
    Check the configuration of your searchkit block!
  </div>
  <i />
</div>
`;
